### ISSUE TITLE
uninitialized class variable @@schemes in URI (NameError)

### ISSUE THREAD

**Issue by marek22k** [ananace/ruby-matrix-sdk#28]
```
$irb
3.1.1 :001 > require 'matrix_sdk'
/home/bandura/.rvm/gems/ruby-3.1.1/gems/matrix_sdk-2.5.0/lib/matrix_sdk/util/uri.rb:15:in `<module:URI>': uninitialized class variable @@schemes in URI (NameError)
Did you mean?  scheme_list
	from /home/bandura/.rvm/gems/ruby-3.1.1/gems/matrix_sdk-2.5.0/lib/matrix_sdk/util/uri.rb:5:in `<top (required)>'
	from <internal:/home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /home/bandura/.rvm/gems/ruby-3.1.1/gems/matrix_sdk-2.5.0/lib/matrix_sdk.rb:4:in `<top (required)>'
	from <internal:/home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:160:in `require'
	from <internal:/home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:160:in `rescue in require'
	from <internal:/home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:149:in `require'
	from (irb):1:in `<main>'
	from /home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
	from /home/bandura/.rvm/rubies/ruby-3.1.1/bin/irb:25:in `load'
	from /home/bandura/.rvm/rubies/ruby-3.1.1/bin/irb:25:in `<main>'
<internal:/home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- matrix_sdk (LoadError)
	from <internal:/home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from (irb):1:in `<main>'
	from /home/bandura/.rvm/rubies/ruby-3.1.1/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
	from /home/bandura/.rvm/rubies/ruby-3.1.1/bin/irb:25:in `load'
	from /home/bandura/.rvm/rubies/ruby-3.1.1/bin/irb:25:in `<main>'

$ruby -v        
ruby 3.1.1p18 (2022-02-18 revision 53f5fc4236) [x86_64-linux]
```

installed ruby via rvm

**Comment by ananace** [comment]
The issue is fixed in the latest master, but I haven't had time to cut a release in a bit. I'll see about getting another version out the door as soon as I can.

**Comment by pinnymz** [comment]
Ugly, but for anyone blocked by this, the following code works in a pinch (must be called before requiring the gem):

```
require 'uri'

module URI
  module LegacySchemesAdapter
    def self.[]=(key, val)
      URI.register_scheme key, val
    end

    def self.key?(key)
      URI.scheme_list.key?(key)
    end
  end

  @@schemes = LegacySchemesAdapter
end
```

**Comment by nerdinand** [comment]
This doesn't seem to be fixed in the latest master. However, @pinnymz's workaround seems to work... It's unfortunately not a good first impression when already requiring the gem fails...

**Comment by ananace** [comment]
I ran into some unrelated issues with older rubies, hence why the release has been delayed a bit, but all my testing on master on the current URI handling has been successful - all the way from 2.6 to 3.1

If this is still broken on master for you then I'll need more logs - and what Ruby version you're seeing the issue on, because it _should_ work.

**Comment by ananace** [comment]
Finally had the time to finish my own debugging and cut the release, so the latest release should include the Ruby 3.1 URI fixes - as well as some assorted Ruby 3.0+ improvements in more edge-case areas.
